### ISSUE TITLE
"Sort by Name" in browser does not trigger sync

### ISSUE THREAD

**Issue by jdevl32** [xbrowsersync/app#233]
When using "Sort by Name" in the browser bookmark manager, it does not trigger a sync up (in the case of the order being affected).  Further, this usually causes errors some short time afterwards if continuing to make changes.  For instance, suppose in some bookmark folder I have (in original order):
B (folder)
E (bookmark)
A (folder)
C (bookmark)
D (folder)

If I then do "Sort by Name", the new order should be:
A (folder)
B (folder)
D (folder)
C (bookmark)
E (bookmark)

Folders are listed first in alphabetical order, followed by bookmarks in alphabetical order.  I notice that a sync up is not triggered.  If, suppose, I then go into folder B and make changes to bookmarks, a sync error usually occurs with a popup instructing me to redo the last unsynced changes.  This usually attempts to reset the original order and structure, but sometimes gets things mangled.

I've seen this behavior in Chrome, and have not tested in any other browser.  I have noticed that if I manually reorder (not using "Sort by Name", but by moving the folders and bookmarks myself), things work as expected -- I see that a sync up occurs as it should.

**Comment by nero120** [comment]
@jdevl32 nice spot. Using "sort by name" in the Chrome bookmarks manager does not trigger moved events in the web extension API, so xBrowserSync would not know about the sort or fire new syncs. Looks like this is a bug with Chromium but I've not found an existing bug [logged](https://bugs.chromium.org/p/chromium/issues/list?q=sort%20bookmarks%20component%3DPlatform%3EExtensions&can=2) that references this issue yet. I will create one but I wouldn't hold my breath in it getting addresses any time soon (usually takes years if it's not a high priority which this won't be).

However on the bright side, the upcoming v1.6.0 release changes the way native bookmarks are linked to synced bookmarks so you won't get any sync errors if you modify existing bookmarks after using the sort by name functionality.

I'll close this as there's not much more I can do until the chromium team fixes the issue.

**Comment by nero120** [comment]
@jdevl32 after doing a bit more investigation I can see that the bookmarks API has a different event [`onChildrenReordered`](https://developer.chrome.com/extensions/bookmarks#event-onChildrenReordered) which I missed. This event is triggered when you use the sort by name option in the bookmarks manager, so I'll implement this for the upcoming release.

**Comment by bu11etpr00f** [comment]
Just wanted to add some info to this. I found this ticket and decided to add to it instead of create a new one.

I have noticed that the reorder no longer causes a Sync Error and a "redownload" of the Bookmarks when done on its' own, but if you reorder a folder (sort by name), and then move an existing bookmark from one folder to another bookmark folder, you will get the Sync Error and "redownload".

I've also noticed that when you try to add a new bookmark, create a new folder, and change the name of the bookmark all in one step; you will get the Sync Error and "redownload" as well. The way I've found to work around this is to:

- add the bookmark, allow it to sync
- change the name (might be able to do this in the first step), allow it to sync
- add the folder without moving the bookmark, allow it to sync
- move the bookmark to the new folder

This might be a different issue, but it seems related.

It would also be nice to "force" a sync even if one is not detected as needed. :)

Hopefully this helps the Dev. AWESOME job BTW! I love xbrowsersync!
